net/src/main/org/apollo/net/codec/login/LoginDecoder.java

```diff
@@ -180,6 +191,7 @@ private void decodePayload(ChannelHandlerContext ctx, ByteBuf buffer, List<Objec
 			String hostAddress = InetAddresses.toAddrString(socketAddress.getAddress());
 
 			if (password.length() < 6 || password.length() > 20 || username.isEmpty() || username.length() > 12) {
+				logger.warning("Username ('" + username + "') or password (length: " + password.length() + ") did not pass validation.");
```

Comment:
Could drop the password length here but it might be helpful.

Reply:
Actually, this could be a potential cause for security concerns (password length being leaked on a typo'd username, for example). I'll remove it for now and just leave the username.